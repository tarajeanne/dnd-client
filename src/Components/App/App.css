@media print {
  .Header, .user-controls, .character-creator-header, .nav-area {
    display: none;
  }
}

main {
  padding-bottom: 30px;
}