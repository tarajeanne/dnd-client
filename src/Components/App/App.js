import React from 'react';
import { Route, Switch } from 'react-router-dom';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service';
import IdleService from '../../services/idle-service';

import Footer from '../Footer/Footer';
import LandingPage from '../../Routes/LandingPage';
import UserPage from '../../Routes/UserPage';
import NotFoundPage from '../../Routes/NotFoundPage';
import CharacterCreator from '../../Routes/CharacterCreator';
import PublicOnlyRoute from '../Utils/PublicOnlyRoute';
import PrivateRoute from '../Utils/PrivateRoute';

import './App.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null
    };
  }

  componentDidMount() {
    IdleService.setIdleCallback(this.logoutFromIdle);

    if (TokenService.hasAuthToken()) {
      IdleService.registerIdleTimerResets();
      TokenService.queueCallbackBeforeExpiry(() => {
        AuthApiService.postRefreshToken();
      });
    }
  }

  componentWillUnmount() {
    IdleService.unRegisterIdleResets();
    TokenService.clearCallbackBeforeExpiry();
  }

  logoutFromIdle = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.forceUpdate();
  };

  render() {
    return (
      <div className="app">
        <main className="App_main">
          {this.state.error && (
            <p className="red">There was an error! Oh no!</p>
          )}
          <Switch>
            <PublicOnlyRoute
              exact
              path={['/', '/login', '/register']}
              component={LandingPage}
            />
            <PrivateRoute path={'/user'} component={UserPage} />
            <PrivateRoute
              path={'/character/:characterId'}
              component={CharacterCreator}
            />
            <Route component={NotFoundPage} />
          </Switch>
        </main>
        <Footer />
      </div>
    );
  }
}

export default App;
