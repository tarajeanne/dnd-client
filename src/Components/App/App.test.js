import React from 'react';
import App from './App';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('App', () => {
    it('renders as expected', () => {
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
  })