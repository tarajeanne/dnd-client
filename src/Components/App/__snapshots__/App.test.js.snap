// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<div
  className="app"
>
  <main
    className="App_main"
  >
    <Switch>
      <PublicOnlyRoute
        component={[Function]}
        exact={true}
        path={
          Array [
            "/",
            "/login",
            "/register",
          ]
        }
      />
      <PrivateRoute
        component={[Function]}
        path="/user"
      />
      <PrivateRoute
        component={[Function]}
        path="/character/:characterId"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
  <Footer />
</div>
`;
